pkg/manifests/manifests.go

```diff
@@ -0,0 +1,150 @@
+		}
+
+		combinedManifestPaths = append(combinedManifestPaths, downloadedManifestPaths...)
+	}
+
+	localManifestSrcDir := filepath.Join(ctx.ImageConfigDir, "kubernetes", "manifests")
+	localManifestPaths, err := getLocalManifestPaths(localManifestSrcDir)
+	if err != nil {
+		return nil, fmt.Errorf("error getting local manifest paths: %w", err)
+	}
+
+	combinedManifestPaths = append(combinedManifestPaths, localManifestPaths...)
+
+	for _, manifestPath := range combinedManifestPaths {
+		manifestData, err := readManifest(manifestPath)
+		if err != nil {
+			return nil, fmt.Errorf("error reading manifest %w", err)
+		}
+
+		storeManifestImageNames(manifestData, extractedImagesSet)
+	}
+
+	for _, definedImage := range ctx.ImageDefinition.EmbeddedArtifactRegistry.ContainerImages {
+		extractedImagesSet[definedImage.Name] = definedImage.SupplyChainKey
+	}
+
+	allImages := make([]image.ContainerImage, 0, len(extractedImagesSet))
+	for imageName, supplyChainKey := range extractedImagesSet {
+		containerImage := image.ContainerImage{
+			Name:           imageName,
+			SupplyChainKey: supplyChainKey,
+		}
+		allImages = append(allImages, containerImage)
+	}
+
+	return allImages, nil
+}
+
+func readManifest(manifestPath string) (any, error) {
+	manifestData, err := os.ReadFile(manifestPath)
+	if err != nil {
+		return nil, fmt.Errorf("error reading manifest: %w", err)
+	}
+
+	if len(manifestData) == 0 {
+		return nil, fmt.Errorf("invalid manifest")
+	}
+
+	var manifest any
+	err = yaml.Unmarshal(manifestData, &manifest)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshalling manifest yaml '%s': %w", manifestPath, err)
+	}
+
+	return manifest, nil
+}
+
+func storeManifestImageNames(data any, imageSet map[string]string) {
+
+	var findImages func(data any)
+	findImages = func(data any) {
+		switch t := data.(type) {
+		case map[string]any:
+			for k, v := range t {
+				if k == "image" {
+					if imageName, ok := v.(string); ok {
+						imageSet[imageName] = ""
+					}
+				}
+				findImages(v)
+			}
+		case []any:
+			for _, v := range t {
+				findImages(v)
+			}
+		}
+	}
+
+	findImages(data)
+}
+
+func getLocalManifestPaths(src string) ([]string, error) {
+	if src == "" {
+		return nil, fmt.Errorf("manifest source directory not defined")
+	}
+
+	var manifestPaths []string
+
+	manifests, err := os.ReadDir(src)
+	if err != nil {
+		return nil, fmt.Errorf("reading manifest source dir: %w", err)
+	}
+
+	for _, manifest := range manifests {
+		manifestName := strings.ToLower(manifest.Name())
+		if filepath.Ext(manifestName) != ".yaml" && filepath.Ext(manifestName) != ".yml" {
+			zap.S().Warnf("Skipping %s as it is not a yaml file", manifest.Name())
+			continue
+		}
+
+		sourcePath := filepath.Join(src, manifest.Name())
+		manifestPaths = append(manifestPaths, sourcePath)
+	}
+
+	return manifestPaths, nil
+}
+
+func downloadManifests(ctx *image.Context, destPath string) ([]string, error) {
```

Comment:
This function doesn't require a full context. If it accepts just the slice of manifests, then all the `setupContext()` calls will not be necessary in the tests.